### DISCUSSION TITLE
Any example creating multiple resources in loop?

### DISCUSSION THREAD

**Discussion by dzmitry-lahoda** [terranix/terranix#69] [Q&A] [answered]
I want to create 3 static IPs and then attach to 3 VMs.

**Comment by mrVanDalo** [comment, marked as answer]
You can for example use the `count` variable of terraform. I'm not sure about the `foreach` keywords of HCL, must try first. But of course you can use the nix way to do it.

```nix
resource = mapAttrs (name: instance_type: {
  aws_instance."example_${name}" = {
    name = "example_${name}";
    instance_type = instance_type;
  };
  aws_eip."example_${name}" = {
    instance = "\${aws_instance.example_${name}}";
  };
}) { 
  primary = "t2.xlarge"; 
  replica01 = "t2.large"; 
  replica02 = "t2.large"; 
};
```

**Comment by mrVanDalo** [comment]
You can check out : 
https://terranix.org/documentation/functions/ and https://terranix.org/documentation/terranix-vs-hcl/ for more information.
